### DISCUSSION TITLE
Are issues regarding uBlock Origin's interaction with other extensions generally accepted?

### DISCUSSION THREAD

**Discussion by partingscientist** [uBlockOrigin/uBlock-issues#2359] [Q&A] [answered]
If yes, would there be any prerequisite(s) specific to such cases that need to be met before reporting? I'd assume one of them (assuming they're allowed) would be such issues must be confirmed caused by (changes to) uBlock Origin. I thought I'd ask first before wasting my time exploring further.

Thank you.

**Comment by gorhill** [comment, marked as answer]
It depends? What's the actual issue?

**Reply by partingscientist** [reply]
Since 1.44.0, users who installed the popular add-on [Dark Reader](https://addons.mozilla.org/en-US/firefox/addon/darkreader) cannot use the DOM inspector due to the page being covered in a dark non-transparent frame (not quite sure what's the appropriate terminology) upon usage.

**Reply by gorhill** [reply]
What browser is this?

**Reply by peace2000** [reply]
I have noticed this issue as well but haven't paid attention if there's a specific version that is related to this issue.

If I have Dark Reader activated on https://www.iltalehti.fi/ and I try to open DOM inspector, the page looks like this:

![kuva](https://user-images.githubusercontent.com/17256841/201703003-a624cbd9-5dd3-4a25-b8b3-e5a16524313b.png)

I have noticed this on other pages as well but not on every page, I think. Running Firefox 106.0.5 and uBO 1.45.3b0.

(Edit, I think it's after all every page where Dard Reader is active).

**Reply by peace2000** [reply]
https://codebeautify.org/sort-text-lines

![kuva](https://user-images.githubusercontent.com/17256841/201704044-c84afb57-f7b5-4e6c-a8b8-fe549e7fe800.png)

**Reply by gorhill** [reply]
Removing CSS `color-scheme` property seems to fix the issue. I just feel stupid for not being able to understand why that is.

https://github.com/gorhill/uBlock/commit/e68537513f0e95a52a553d98fa4da78a5f948178

**Reply by partingscientist** [reply]
Regarding Dark Reader, kindly cc-ing @Gusted

**Reply by Gusted** [reply]
Hmm weird, which version was still "buggy"? Not able to reproduce this, anyhow `color-scheme` has caused various bugs for Dark Reader as well, it's really a black box and documentation is really vague on what the implementation should look like in browsers. I feel you @gorhill regarding this.

**Reply by gorhill** [reply]
The thing which is difficult to understand is that `color-theme` works fine with element picker.

**Reply by gorhill** [reply]
Ok I think I understand the issue -- which is [still present](https://github.com/gorhill/uBlock/commit/e68537513f0e95a52a553d98fa4da78a5f948178#commitcomment-90210630). Something I overlooked is that the DOM inspector uses an anonymous (`about:blank`) frame for the layer, while it's an extension (`moz-extension://...`) frame with the element picker. Dark Reader has access to the anonymous frame, which it modifies, while it does not have access to the extension frame. Ideally the DOM inspector iframe would be an extension one, but this is not a trivial change. I think this can be fixed by simply ensuring uBO's style inside the frame wins over Dark Reader one's.

**Reply by Gusted** [reply]
Dark Reader can be quite aggressive about getting priority over other styles. Dark Reader should not add the `color-scheme: dark` property when it's in a `about:blank` frame, WDYT? It's easier to patch this on Dark Reader's side as it's ultimately causing this behavior.

**Reply by gorhill** [reply]
It's fixed in uBO already: <https://github.com/gorhill/uBlock/commit/105c7f54466b5f5f1f5754607d5852a147171802> -- it's added as an element style, which has the highest priority.
